### PR TITLE
Update to Spout2.007f

### PR DESCRIPTION
**Pull request by mzlt**
* Use new SputDX class included with spout2.007 instead of
  spoutSenderNames and spoutDirectX classes
* Move the management of DirectX11 from the plugin to SpoutDX class

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -67,14 +68,35 @@ function(copy_spout_file)
...
+			"${OBS_OUTPUT_DIR}/$<CONFIGURATION>/${OBS_PLUGIN_DESTINATION}/Spout.pdb")
+	add_custom_command(
+		TARGET win-spout POST_BUILD
+		COMMAND ${CMAKE_COMMAND} -E copy
+			"${CMAKE_CURRENT_SOURCE_DIR}/deps/Spout2/BUILD/Binaries/${_src_suffix}/SpoutLibrary.dll"
 			"${OBS_OUTPUT_DIR}/$<CONFIGURATION>/${OBS_PLUGIN_DESTINATION}/SpoutLibrary.dll")
+	add_custom_command(
+		TARGET win-spout POST_BUILD
+		COMMAND ${CMAKE_COMMAND} -E copy
+			"${CMAKE_CURRENT_SOURCE_DIR}/deps/Spout2/BUILD/Binaries/${_src_suffix}/SpoutLibrary.pdb"
```
Comment (campbellwmorgan) [inline CMakeLists.txt:86]: Worried that this will break when there are no `pdb` files (ie when only built for release)

Comment (mzlt) [reply CMakeLists.txt]: Update: It omits pdb's for "Release" now.
1. Should we add something that checks if pdb's are available in "Debug" - else build without
2. or force fail if config is "Debug" and there are no pdb's ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -50,31 +49,59 @@ include_directories(
...
-	add_custom_command(
-		TARGET win-spout POST_BUILD
+			"${CMAKE_CURRENT_SOURCE_DIR}/deps/Spout2/BUILD/Binaries/${_src_suffix}/SpoutLibrary.dll"
+			"${OBS_OUTPUT_DIR}/$<CONFIGURATION>/${OBS_PLUGIN_DESTINATION}/SpoutLibrary.dll"
 		COMMAND ${CMAKE_COMMAND} -E copy
-			"${CMAKE_CURRENT_SOURCE_DIR}/deps/spout/SPOUT_LIBRARY/Binaries/${_src_suffix}/SpoutLibrary.dll"
-			"${OBS_OUTPUT_DIR}/$<CONFIGURATION>/${OBS_PLUGIN_DESTINATION}/SpoutLibrary.dll")
+			"${CMAKE_CURRENT_SOURCE_DIR}/deps/Spout2/BUILD/Binaries/${_src_suffix}/SpoutDX.dll"
+			"${OBS_OUTPUT_DIR}/$<CONFIGURATION>/${OBS_PLUGIN_DESTINATION}/SpoutDX.dll"
+		  COMMAND if $<CONFIG:Debug>==1 (
```
Comment (campbellwmorgan) [inline CMakeLists.txt:78]: Ah nice! You can put if statements in there. Good find :)

File: win-spout-installer.nsi [hunk 3]
```diff
@@ -61,10 +64,14 @@ Section "Spout 2 OBS Plugin" Section1
 	File "..\..\build64\plugins\win-spout\Release\win-spout.dll"
 	File "..\..\build64\plugins\win-spout\Release\win-spout.exp"
 	File "..\..\build64\plugins\win-spout\Release\win-spout.lib"
-	File "..\..\plugins\win-spout\deps\spout\SPOUT_LIBRARY\Binaries\x64\SpoutLibrary.dll"
-	File "..\..\plugins\win-spout\deps\spout\SPOUT_LIBRARY\Binaries\x64\SpoutLibrary.lib"
-	File "..\..\plugins\win-spout\deps\spout\VS2012\Binaries\x64\Spout.dll"
-	File "..\..\plugins\win-spout\deps\spout\VS2012\Binaries\x64\Spout.lib"
+	File "..\..\plugins\win-spout\deps\Spout2\BUILD\Binaries\x64\Spout.dll"
+	File "..\..\plugins\win-spout\deps\Spout2\BUILD\Binaries\x64\Spout.lib"
+	File "..\..\plugins\win-spout\deps\Spout2\BUILD\Binaries\x64\Spout_static.lib"
```
Comment (campbellwmorgan) [inline win-spout-installer.nsi:69]: Not sure we're using this are we? In fact, a while ago I think I realised that the `lib` files aren't even required in the installer, but lets fix that in a later PR

**Review by campbellwmorgan — APPROVED** [review]

**Pull request merged**